fastai/callback/hook.py

```diff
@@ -144,12 +144,26 @@ def total_params(m):
 # Cell
 def layer_info(learn, *xb):
     "Return layer infos of `model` on `xb` (only support batch first inputs)"
-    def _track(m, i, o): return (m.__class__.__name__,)+total_params(m)+(apply(lambda x:x.shape, o),)
+    def _track(m, i, o):
+        for inp, out in zip(i,o):
+            if inp.shape[1:] == out.shape: same = True
+            else: same = False
+        if hasattr(m, 'weight'): # non activation layer
```

Comment:
There's a lot of duplicate code here. Can you please clean it up? Also, I don't think we need to check for weights or param groups do we? We should only be checking whether the output size changes.

Reply:
This let's us check what freeze group they are in (so it's something new). Would there be a better way to check what param groups a layer is in other than checking by weight? I also checked for weights because we discussed differentiating between activation functions, poolings, and everything else. Looking for the weights helps me do this.

Reply:
@jph00 I've simplified this a bit